framework/wazuh/core/cluster/master.py

```diff
@@ -992,6 +1055,63 @@ def to_dict(self) -> Dict:
         return {'info': {'name': self.configuration['node_name'], 'type': self.configuration['node_type'],
                          'version': metadata.__version__, 'ip': self.configuration['nodes'][0]}}
 
+    def get_agent_groups_info(self, client):
+        """Function in charge of ensuring that in each iteration of agent-groups, the updated group information will be sent only one time to each worker node.
+        The variable with this information will not be updated until all the MasterHandlers send the information to their worker node.
+
+        Parameters
+        ----------
+        client : str
+            String with the node name.
+
+        Returns
+        -------
+        result : dict
+            Updated data on agent-groups.
+        """
+        result = {}
+        if client not in self.agent_groups_control_workers:
+            result = self.agent_groups_control
+            self.agent_groups_control_workers.add(client)
+        elif len(self.agent_groups_control_workers) == len(self.clients.keys()):
+            self.agent_groups_control = {}
+            self.agent_groups_control_workers.clear()
```

Comment:
This could cause a race condition. In case that a worker which has received the agent-groups information disconnects, the condition would be met even if there is another MasterHandler that did not yet send the data to its worker, so it would end up outdated.